### PR TITLE
Gradle tests

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1491)

### CODE REVIEW

File: analyzer/src/funTest/kotlin/GradleTest.kt [hunk 1]
```diff
@@ -118,6 +118,10 @@ class GradleTest : StringSpec() {
             patchActualResult(yamlMapper.writeValueAsString(result)) shouldBe expectedResult
         }
 
+        // Disabled because despite following the example at [1] Gradle says there is "No service of type
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/GradleTest.kt]: Commit message nit: "tests ARE disabled"

File: analyzer/src/funTest/kotlin/GradleTest.kt [hunk 2]
```diff
@@ -132,6 +136,7 @@ class GradleTest : StringSpec() {
             yamlMapper.writeValueAsString(result) shouldBe expectedResult
         }
 
+        // Disabled as it causes hangs and memory issues.
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/GradleTest.kt]: Maybe add "... on CI", IIRC we have only seen it there, never locally.

File: analyzer/src/funTest/kotlin/GradleTest.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/funTest/kotlin/GradleTest.kt]: Done.

File: analyzer/src/funTest/kotlin/GradleTest.kt (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/funTest/kotlin/GradleTest.kt]: Done.

**Pull request merged**